### PR TITLE
Support macro for trait items

### LINKED ISSUE #4039: Support macro expansion of trait items
Example case:

```rust
macro_rules! expand {
    ($i:ident) => {
        fn foo() -> usize { 100 }
    };
}
trait Foo {
    expand!(foo);
}
struct Bar;
impl Foo for Bar {}

fn main() {
    let _a = Bar::foo();   // _a should be infer to usize
}
```

### PR DESCRIPTION
**Pull request by edwin0cheng**
Fixed  #4039

r? @flodiebold

### CODE REVIEW

File: crates/ra_hir_def/src/data.rs [hunk 1]
```diff
@@ -150,51 +150,31 @@ pub struct TraitData {
...
-                })
-                .collect()
-        } else {
-            Vec::new()
-        };
+        let mut items = Vec::new();
+
+        if let Some(item_list) = src.value.item_list() {
+            let mut expander = Expander::new(db, tr_loc.ast_id.file_id, module_id);
+            items.extend(collect_impl_items(
```
Comment (matklad) [inline crates/ra_hir_def/src/data.rs:164]: let's remove `impl` from the name of this and following function, as they are no longer impl specific

Comment (edwin0cheng) [reply crates/ra_hir_def/src/data.rs]: Fixed. 

But they way, We named `ImplItem` for all items inside both `ast::ImplDef` and `ast::TraitDef`. Maybe we should rename to something else?

Comment (matklad) [reply crates/ra_hir_def/src/data.rs]: Good observation, we  should rename it to `AssocItem`

**Pull request merged**